### ISSUE TITLE
Minor documentation amendment

### ISSUE THREAD

**Issue by digdilem** [uyuni-project/uyuni#2645]
Following "Server - Minor Upgrade" section of 

https://www.uyuni-project.org/doc/2020.09/uyuni_upgrade_guide.pdf

This did not upgrade on my installation. The Uyuni repo was not automatically refreshed.

Suggestion:

Include an additional step between stage 1. and 2. of 

2. Refresh the uyuni repository

> zypper ref

**Comment by paususe** [comment]
@keichwa 
Could you please implement this change?

**Comment by keichwa** [comment]
Thanks for the report!

**Comment by paususe** [comment]
@keichwa it seems this was added to the docs, so closing the issue.

**Comment by keichwa** [comment]
yes, thanks for closing!

**Comment by juliogonzalez** [comment]
@keichwa this is now reported at the list as missing: https://lists.opensuse.org/archives/list/users@lists.uyuni-project.org/message/6PACXIDHGBZZGUM32LTR6QUBG4YQPU6P/

Can you have a look?

**Comment by keichwa** [comment]
I guess it was removed because we have this note after the procedure:
```
[NOTE]
====
By default, zypper refreshes the repository every ten minutes (see ``repo.refresh.delay`` in [path]``/etc/zypp/zypp.conf``).
If ``autorefresh`` is disabled, run [command]``zypper ref`` to refresh all repositories.
====
```
I think it would be save if I'd add `zypper ref` as an OPTIONAL step, and move the note to a place before the procedure.  @juliogonzalez would this work?

**Comment by digdilem** [comment]
I think the problem that I (and, I guess, Heiner) encountered is that
autorefresh is disabled on OpenSuse by default - or at least, was when I
built our uyuni box some time ago.

Given that 'zypper ref' does no harm when autorefresh *is* enabled, and
allows the upgrade to not fail when it's *not *enabled, my suggestion is to
keep it in as a full step.

If there's resistance to this, then perhaps couch that step with "(This
step may be skipped if the system is automatically refreshing Zypper's
repositories at a frequent interval)" or similar?

**Comment by juliogonzalez** [comment]
@digdilem @keichwa I agree with just instructing the user to call `zypper ref` as a "full step", meaning we will not really need a note.

@keichwa is there any particular reason you want to have it as optional and keep the note?

**Comment by keichwa** [comment]
Julio González Gil ***@***.***> writes:

> @digdilem @keichwa I agree with just instructing the user to call
> `zypper ref` as a "full step", meaning we will not really need a note.
>
> @keichwa is there any particular reason you want to have it as
> optional and keep the note?

The glorious details are here:
https://github.com/uyuni-project/uyuni-docs/pull/584

Together with digdilem's background info I think that we should treat it
differently for Uyuni (add 'zypper ref' step) and SUMA (note is enough
because autorefresh is be default enabled).

-- 
Karl Eichwalder

SUSE Software Solutions Germany GmbH, GF: Felix Imendörffer
HRB 36809, AG Nürnberg  Maxfeldstr. 5,  D-90409 Nürnberg

**Comment by juliogonzalez** [comment]
I see mlandres was pretty clear there:

> So if you know all enabled repos have autorefresh on, you don't need an explicit refresh. Otherwise you do (unless you know the repos do not contribute to the command).

But even if we specify `zypper ref` for SUSE Manager as well... what's exactly the problem? The refresh does not download all the metadata if it didn't change, it only checks if it needs to download it.

**Comment by keichwa** [comment]
Yes, agreed.  The step is back: https://github.com/uyuni-project/uyuni-docs/pull/1213
I'll keep the note nevertheless; if we want to remove it, we can create a separate PR.
